manifests/profiles/infra/capitpls/aws.yaml.tpl

```diff
@@ -0,0 +1,149 @@
+apiVersion: cluster.x-k8s.io/v1beta1
```

Comment:
It is weird why donot you translate to golang code.

With yaml. it is a black box to developer and difficult to do test.  The indents spaces and even contents are not easy to maintain

Reply:
IMO, use a template is easier than golang code.

`reconcileAWSCluster` handle the reconcile logic of `AWSCluster`, which cannot do in this way.